default/python/stub_generator/parse_docs.py

```diff
@@ -110,16 +111,37 @@ def parse_name(txt):
     return [x[0] for x in args], return_type
 
 
-def merge_args(arg_types, is_class):
+def merge_args(name, raw_arg_types, is_class):
+    """
+    Merge multiple set of arguments together.
+
+    Single argument set is used as is.
+    If we have two unique argument sets, and on of them is empty, use keywords.
+    In other cases log error and use first one.
+
+    :param str name:
+    :param list[tuple] raw_arg_types:
+    :param bool is_class:
+    :rtype: (list[str], list[(str, str)])
+    """
+    # If wrapper define functions that have same name, and same arguments but different return types,
+    # it will come here with len(arg_types) >= 2, where all arguments set are the same.
+    size = len(raw_arg_types)
+    arg_types = sorted(set(raw_arg_types))
+    if len(arg_types) != size:
+        error("[%s] Duplicated argument types", name)
```

Comment:
I'm not understanding why it should be an error if two of the arguments have the same type-- could you explain?

Reply:
`arg_types` is a list of argument tuples when method overloaded.

Example message when method have two sets of arguments and they are different.
```
[canBuild] Cannot merge, use first argument group from:
(empire)arg1, (buildType)arg2, (str)arg3, (int)arg4
(empire)arg1, (buildType)arg2, (int)arg3, (int)arg4
```

This line prevents crash when they all overloads have same signature.
This can happen when there is an error in wrapper with a duplicated declaration.
Old code also crashed in case of `arg_types= [(), ()]`